File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
import sys
input = sys.stdin.readline
while True:
    a, b, c = map(int, input().split())

    if a == 0 and b == 0 and c == 0:
        sys.exit(0)

    temp = max(a, b, c)

    if temp >= (a + b + c - temp):
        print('Invalid')
    else:
        if a == b == c:
            print('Equilateral')
        elif a != b and b != c and a != c:
            print('Scalene')
        else:
            print('Isosceles')

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.c
#include <stdio.h>

int max(int a, int b, int c) {
    int temp = -1000;

    if (a > temp) {
        temp = a;
    }

    if (b > temp) {
        temp = b;
    }

    if (c > temp) {
        temp = c;
    }

    return temp;
}

int main() {

    int a, b, c;

    do {
        
        scanf("%d %d %d", &a, &b, &c);

        if (a == 0 && b == 0 && c == 0) {
            break;
        }

        int max_num = max(a, b, c);

        if (max_num >= (a + b + c - max_num)) {
            printf("Invalid\n");
        }
        else if (a == b && b == c && a == c) {
            printf("Equilateral\n");
        }
        else if (a != b && b != c && a != c) {
            printf("Scalene\n");
        }
        else {
            printf("Isosceles\n");
        }

    } while (1);

    return 0;
}